### PR TITLE
Add script to check router config

### PR DESCRIPTION
**Pull request by lulf**
@grs Please review. I hope to use this script either as part of a
liveness probe or the smoketest to verify router configs

### CODE REVIEW

File: qdrouterd/router-check.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    return client.response
+ 
+def verify_router_config(address_config, router_config):
+    for address in address_config:
+        address_definition = address_config[address]
+        if address_definition["store_and_forward"] == True and address_definition["multicast"] == False:
+            found = False
+            results = router_config["results"]
+            for entry in results:
+                name = entry[0]
```
Comment (grs) [inline qdrouterd/router-check.py]: Is there supposed to be a test here? I.e. does the specific address occur in the results?

Comment (lulf) [inline qdrouterd/router-check.py]: At line 55, it will throw an exception if it haven't found the queue in the config.

**Comment by grs** [comment]
I'm a bit conflicted on this one...

For a liveness probe, I would do something simpler (i.e. just verify that we can connect via AMQP). Tying liveness to synchronized config seems too strict to me. It could result in a router being restarted unnecessarily  if the probe is fired just after (or during) a change to the config.

I also feel that the verification is in some ways duplicating what the router agent does. In itself that is not necessarily bad, but it means that whenever we update the router agent to handle some new logic, we would need to make the same changes to this script.

I think my instinct would be to add logic to the router agent allowing it to be queried as to the current synchronization of the routers. That would then be something that could be used more easily programmatically (i.e. send a request and get back a response indicating status, or when synchronized).

That said, a separate check (i.e. independent of the agent) would also avoid having any bug in the agent affect the result (of course you could have a bug in the  independent check as well, but it is simpler so less space to hide).

**Comment by lulf** [comment]
Sorry, I used the wrong term 'liveness' probe, while I was actually experimenting with using it as a 'readinessProbe'. Failing the readiness check only causes it to be "out of rotation", not restarted. Maybe this is too strict as well. Though my original plan was to invoke it manually in the smoketest, so I can keep that scope and drop the readiness probe.

I agree that having it as part of the router agent would actually be better. This started out as a simple script to use in the smoketest, and then I thought it might be used more generically. I can try to add a simple handler for it in the router agent. 

I really like the idea of having a 'reply when synchronized' in the router agent or something like that, though it would require the 'desired' configuration state to be provided by the requester to work appropriately.

**Comment by grs** [comment]
Ah, ok, as a readiness probe that would indeed a reasonable check.

I'm fine with the change. Adding the ability to get current synchronization status from the router agent could be an additional thing. Regarding the 'desired' configuration state, the request could contain an optional address to verify. The check would then ensure that (a) that address matched the agents understanding and (b) all the routers were in sync with the latest update from configuration service.

**Comment by lulf** [comment]
Great, I'll merge this for now so that it can be used as an easy debugging tool until we have something better.

**Pull request merged**